### ISSUE TITLE
Future: Web stack

### ISSUE THREAD

**Issue by ts** [mkwmms/prazie#1]
Let's start talking about the web stack so we can all orbit the same planet. Initial thoughts? I know we need to define the web/native boundaries…

**Comment by mwilliammyers** [comment]
I'm game with anything. 

I'm not super familiar with front end but in terms of backend I saw a sweet new [swift server library](https://github.com/PerfectlySoft/Perfect) but swift is probably not mature enough in that realm, but could be cool. Thats probably just me wishing haha. 

I'm most comfortable with Ruby and Python though. Although, I haven't explicitly done anything in rails or flask... Well I also know Java but I don't think we want to go that route...

**Comment by ts** [comment]
I saw that earlier and wanted to love it, but there isn't anything there yet—just half-baked marketing. There will be a Swift server lib soon that rules, but I think we can level up a bit. I think PHP is friendly enough for the most novice engineer and then we'll have our fair share of Javascript to play with for the web version of the app.

**Comment by ts** [comment]
So, start looking at Laravel. And Laracasts which will get you up to speed quick. I think this makes sense for the site itself and most of the user related CRUD. I'd actually love to build tiny modules/apps in other languages/frameworks and test them against each other.

**Comment by mwilliammyers** [comment]
> I'd actually love to build tiny modules/apps in other languages/frameworks and test them against each other.

Yeah me too! That would be really interesting to see which came out on top. It might even be fun to do it in C++ or Java just to see if it really is _that_ much faster.

> I saw that earlier and wanted to love it, but there isn't anything there yet—just half-baked marketing.

Yeah, I didn't really look at it but that does not surprise me in the slightest. It's a shame though, that would have been fun to do in Swift.

**Comment by tannerlinsley** [comment]
Golang micro service architecture on the backend and a modular angular or
react front end.

**Comment by mwilliammyers** [comment]
> Golang micro service architecture on the backend and a modular angular or react front end.

Yeah I am pretty interested in Golang, it seems pretty legit. And I am doing a project in react at school. It seems pretty cool but Javascript/HTML/front end really isn't my cup o tea...
